File: Backend/LogisticsPlatform.Application/DTOs/Customers/Addresses/CustomerAddressDto.cs
using LogisticsPlatform.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticsPlatform.Application.DTOs.Customers.Addresses
{
    public class CustomerAddressDto
    {
        public string? AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public string? City { get; set; }
        public string? State { get; set; }
        public string? Country { get; set; }
        public string? PostalCode { get; set; }
        public CustomerAddressType? Type { get; set; }
        public bool? IsPrimary { get; set; }
        public bool? IsActive { get; internal set; }

    }
}

File: Frontend/src/app/core/models/customers/addresses/customer-address.dto.ts
import { CustomerAddressType } from "./customer-address-type";

export interface CustomerAddressDto {
    id: string;
    customerId: string;
    addressLine1: string;
    addressLine2?: string;
    city: string;
    state?: string;
    country: string;
    postalCode?: string;
    type: CustomerAddressType;
    isPrimary: boolean;
    isActive: boolean;
}
